src/edifact/parser.ts

```diff
@@ -0,0 +1,172 @@
+import { Interchange, Message, Segment, Element, ServiceStringAdvice } from "./types";
+
+/** Parses an EDIFACT interchange.
+ *  
+ *  The parser does not support groups (UNG, UNE), nesting and only supports syntax UNOC version 3,
+ *  so, requires the input to be encoded with ISO 8859-1.
+ *  It would not be much effort to implement these each, but they are not necessary in this project.
+ * 
+ *  The parser also does not validate the interchange, it assumes that the input is well-formed.
+ * 
+ *  The implementation is based on the documentation from
+ *
+ *  https://www.gs1.org/docs/EDI/eancom/2012/ean02s3/part1/part1_05.htm
+ */
+export default function parse(str: string): Interchange {
+    // first line decides which separators are to be used in the rest of the interchange
+    const ssa = parseServiceStringAdvice(str)
+    const segments = split(str, ssa.segmentTerminator, ssa.releaseCharacter)
+
+    let header = null
+    const messages: Message[] = []
+
+    let messageHeader = null
+    let messageSegments: Segment[] = []
+
+    for (let i = 0; i < segments.length; ++i) { 
```

Comment:
I avoid for loops on arrays for more readability and less verbosity
```suggestion
    segments.forEach((segment, i) => {
```

Reply:
usually a good idea, but not in this case, tried it out (segment needs to be modified anyway, plus more importantly, type inferrence in typescript has some hiccups then for the variable `messageHeader`)